### DISCUSSION TITLE
Idea list

### DISCUSSION THREAD

**Discussion by tripleo1** [jnode-revisited/jnode#3] [Ideas]
See my ideas/roadmap here:

https://github.com/jnode-revisited/jnode/projects/1

Comments welcome

**Comment by ghost** [comment]
Hi tripleo1.

> See my ideas/roadmap here:
> https://github.com/jnode-revisited/jnode/projects/1

Great work!

> Comments welcome

1. So... I was thinking about this idea [cdn for jnode?](https://github.com/jnode-revisited/jnode/discussions/7) *(the general idea is to have a package and dependency manager in jnode, in addition, to have support for protocols like IPFS, Hypercore in jnode, and beyond that, to have a server where packages and jnode dependencies are stored and shared)*
2. A random idea I had today would be to install jnode on wsl(windows subsystem linux) or have wsjnode(windows subsystem jnode).
3. jnode and ui frameworks
4. jnode-webapp
5. jnode-events: meetups about jnode
6. jnode-conference
7. jnode-webview
8. jnode-jobs: companies that hire programmers to create software in jnode
9. jnode certification: certify developers and companies that use jnode
10. jnode training: take courses and teach people how to use jnode
11. Portability of Linux, NodeOS, Haiku applications on jnode
12. jnode-matrix: matrix protocol version in jnode system
13. jnode-gitbug: version [git-bug](https://github.com/MichaelMure/git-bug) in jnode, git-bug is distributed, offline-first bug tracker embedded in git, with bridges.
14. jnode-sqlite: sqlite version for jnode applications
15. jnode-hub: site that brings together visual components as a web component in jnode applications
16. jnode-rqlite: version rqlite in jnode, rqlite is lightweight, distributed relational database built on SQLite
17. jnode-usql: version [usql](https://github.com/xo/usql) in jnode, usql is universal command-line interface for SQL databases. My idea is to have database replication support with the jnode system
18. jnode-ui: Flexible, distributed user interface for any jnode application
19. jnode partners: open source applications that use jnode in backend or companies that finance jnode
20. [jnode-revisited.github.io with Gitflow, GitHub flow or any-flow? ](https://github.com/jnode-revisited/jnode/discussions/5)
21. docker-jnode: version of docker with jnode for any type of application
22. jnode-cli: create apps in jnode with nodejs "create-jnode-app"
23. jnode version on raspberry pi
24. jnode-hosted: host web pages and portals for jnode applications
25. jnode-storage: pod storage(from protocol solid-project ) for jnode clients or jnode applications
26. jnode-store: unhosted, self-hosted, nocode, lowcode jnode apps
27. jnode-rfc: group responsible for research and implementation of the jnode system, documentation
28. jnode-support: I think about the idea of jnode support for clients and applications that run jnode
29. jnode-database-os: lightweight version of jnode focused on big data or smart lake, artificial intelligence for data from small, medium and large companies
30. mitosis & jnode: create apps in different types of graphical interface and in different operating systems
31. jnode settings on a cdn network
32. A complementary idea would be to add a vdi image (virtual disk image) in [osboxes.org](https://www.osboxes.org/) - example: ``https://www.osboxes.org/jnode/`` or ``https://archiveos.org/``
33. development of own hardware for jnode, not depending on hardware from external manufacturers
34. compare features of jnode operating system with JX-os, JavaOS, Haiku, Node-OS, SavaJe OS, Linux, Inferno-os, Redox-OS, Windows, MacOS etc?
35. jnode-channel in youtube, medium, blogger, twitter, mastodon or peertube etc
36. redesign "design" in jnode

*the list is not complete but it's an idea I have for jnode thing*

**Reply by tripleo1** [reply]
Alot of this is whats called pie in the sky thinking.  We need to concentrate on the basics.

These are all (mostly?) good ideas, but I think it's not time for a lot of them

Companies are focused on Linux right now, so DON'T look for financing (unless you know where to find it!)

I like ideas 6, 7 and 8.

I also like the idea of IPFS, and I had never heard of Hypercore, so thanks for that.

Mostly, we need a bootstrap.

**Reply by ghost** [reply]
Hi tripleo1.

> Alot of this is whats called pie in the sky thinking. We need to concentrate on the basics.
> These are all (mostly?) good ideas, but I think it's not time for a lot of them
> Companies are focused on Linux right now, so DON'T look for financing (unless you know where to find it!)
> I like ideas 6, 7 and 8.
> I also like the idea of IPFS, and I had never heard of Hypercore, so thanks for that.
> Mostly, we need a bootstrap.

thank you for feedback.

**Comment by tripleo1** [comment]
Upon review, I like 1, 11, 15, 18. Maybe 16.

Number one might be a good project to start with as there is (a little) java/ipfs around, last time I checked.

This is one of my favorites, although it is in JavaScript: https://github.com/vadimdemedes/ink

**Comment by ghost** [comment]
@tripleo1 how are you? 

Can we put all these ideas into GitHub Issues and archive the ideas that are difficult to do?

**Reply by tripleo1** [reply]
Mabe hfsl

**Reply by tripleo1** [reply]
I'm not typing all that.

I like 1 and 28, see you next year, I guess.
